src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -362,7 +362,14 @@ internal void SetBufferSizes(int readBufferSize, int writeBufferSize)
         public override void Flush()
         {
             if (_handle == null) InternalResources.FileNotOpen();
-            Interop.Termios.TermiosDiscard(_handle, Interop.Termios.Queue.AllQueues);
+
+            SpinWait sw = new SpinWait();
+            while (!(_readQueue.IsEmpty && _writeQueue.IsEmpty))
+            {
+                sw.SpinOnce();
+            }
+
+            Interop.Termios.TermiosDrain(_handle);
```

Comment:
I read about [tcdrain](https://en.wikibooks.org/wiki/Serial_Programming/termios#tcdrain) an [flush](https://en.wikibooks.org/wiki/Serial_Programming/termios#tcflush), and I am wondering if there would be a case where, after calling drain (waiting for all data to finish sending) we would need to do a flush (discard unsent data). 

What if we send all the data, then a short moment passes, and right before the Flush method finishes, we receive more data to send? What would happen to it?

Reply:
@carlossanlop IIRC we talked about this with @wfurt and because this would have to happen in parallel this is not a concern (since we don't promise this is thread-safe class).

Having said that in such case you could cause waiting for the writeQueue to be empty can take a while in such case